gridstatus/spp.py

```diff
@@ -272,6 +302,348 @@ def get_interconnection_queue(self, verbose=False):
+            if verbose:
+                print(f"Fetching {url}", file=sys.stderr)
+            csv = requests.get(url)
+            df = pd.read_csv(io.StringIO(csv.content.decode("UTF-8")))
+            all_dfs.append(df)
+        return pd.concat(all_dfs)
+
+    def _fs_get_dam_lmp_by_location_paths(self, date, verbose=False):
+        """Lists files for Day-ahead Market (DAM), Locational Marginal Price (LMP) by Settlement Location (SL)"""
+        paths = []
+        if date == "latest":
+            raise ValueError("DAM is released daily, so use date='today' instead")
+        elif not utils.is_today(date, self.default_timezone):
+            raise NotSupported("Historical DAM data is not supported currently")
+
+        date = pd.Timestamp.now(
+            tz=self.default_timezone,
+        ).normalize()
+        # list files for this month
+        files_df = self._file_browser_list(
+            name=FS_DAM_LMP_BY_LOCATION,
+            fs_name=FS_DAM_LMP_BY_LOCATION,
+            type="folder",
+            path=date.strftime("/%Y/%m/By_Day"),
+        )
+        max_name = max(files_df["name"])
+        max_file = files_df[files_df["name"] == max_name]
+        # get latest file
+        paths = max_file["path"].tolist()
+
+        if verbose:
+            print(f"Found {len(paths)} files for {date}", file=sys.stderr)
+        return paths
+
+    def _get_marketplace_session(self) -> dict:
+        """
+        Returns a session object for the Marketplace API
+        """
+        html = requests.get(MARKETPLACE_BASE_URL)
+        jsessionid = html.cookies.get("JSESSIONID")
+        soup = BeautifulSoup(html.content, "html.parser")
+        csrf_token = soup.find("meta", {"id": "_csrf"}).attrs["content"]
+        csrf_token_header = soup.find("meta", {"id": "_csrf_header"}).attrs["content"]
+
+        return {
+            "cookies": {"JSESSIONID": jsessionid},
+            "headers": {
+                csrf_token_header: csrf_token,
+            },
+        }
+
+    def _file_browser_list(self, name: str, fs_name: str, type: str, path: str):
+        """Lists folders in a browser
+
+        Returns: pd.DataFrame of files, or empty pd.DataFrame on error"""
+        session = self._get_marketplace_session()
+        json_payload = {"name": name, "fsName": fs_name, "type": type, "path": path}
+        list_results = requests.post(
+            FILE_BROWSER_API_URL,
+            json=json_payload,
+            headers=session["headers"],
+            cookies=session["cookies"],
+        )
+        if list_results.status_code == 200:
+            df = pd.DataFrame(list_results.json())
+            return df
+        else:
+            return pd.DataFrame()
+
+    def _file_browser_download_url(self, fs_name, params=None):
+        qs = "?" + urlencode(params) if params else ""
+        return f"{FILE_BROWSER_API_URL}download/{fs_name}{qs}"
+
+    def _log_url(self, message, url, params={}):
```

Comment:
is this function used? seems helpful but lets remove if not used

Reply:
Good catch